src/Features/CSharp/Portable/CodeFixes/HideBase/HideBaseCodeFixProvider.cs

```diff
@@ -54,7 +55,10 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
                 return;
             }
 
-            context.RegisterCodeFix(new AddNewKeywordAction(context.Document, originalNode), context.Diagnostics);
+            if (diagnostic.Id == CS0108)
+                context.RegisterCodeFix(new AddNewKeywordAction(context.Document, originalNode), context.Diagnostics);
+            else if (diagnostic.Id == CS0109)
+                context.RegisterCodeFix(new RemoveNewKeywordAction(context.Document, originalNode), context.Diagnostics);
```

Comment:
we shuld check to make sure we can *find* teh `new` modifier first before registering the action.  that way there is no way to fail when teh action runs.

Reply:
@CyrusNajmabadi should a SyntaxReference be passed to the fixer or should the logic be called both before registering the fix and while applying the fix?

Reply:
easiest is just call the logic both times.  it's simple and cheap enough to do.

Reply:
Thanks, got it. One question: is it safe to pass `SyntaxNode` (`MemberDeclarationSyntax`) reference to `DocumentChangeAction` delegate or I should retrieve it one more time from `Document` inside the actual code fix?

Reply:
technically it is discouraged.  but about 80% of our own fixers violate that recommendation, so there's no harm in one more doing it :)

Reply:
Sorry @CyrusNajmabadi didn't get it :) By "one more doing it" you mean retrieving `SyntaxNode` two times (before registering codefix and inside a codefix) or do it one time and pass the reference into a codefix?

Reply:
Either :)